### ISSUE TITLE
Passwort reset link on custom account edit page is not working

### ISSUE THREAD

**Issue by frostblogNet** [amzn/amazon-payments-magento-2-plugin#526]
#### What I expected

As a logged in user, I want to go to the page custom account edit to reset my password. If I click on the link  "Request a password to change your account password." I want to get a mail to reset my password.

#### What happened instead
If I click this link I will get redirected to customer account and no mail will be send out.

#### Steps to reproduce the issue

- Create a magento user account using amazon login
- login to that account go to custom account edit page and klick on reset password.
- Click the link "Request a password to change your account password."
- Check you mail box. No mail.

#### Your setup

* Magento version: 2.3.2
* Amazon Pay Extension Version: 3.2.9
* Magento Edition: Commerce

The reson for this issue is that magento is checking in the conntroller vendor/magento/module-customer/Controller/Account/ForgotPassword.php 

```php
if ($this->session->isLoggedIn()) {
            /** @var \Magento\Framework\Controller\Result\Redirect $resultRedirect */
            $resultRedirect = $this->resultRedirectFactory->create();
            $resultRedirect->setPath('*/*/');
            return $resultRedirect;
        }
```
This test is true and therfore no password forgot mail will be send out.

Hope this will help you.

Regards

Jens

**Comment by tarishah** [comment]
thanks @frostblogNet for bringing this to our attention. We will provide an update here as soon as we have the fix. 

Thanks again!
Tarini

**Comment by RayAnaya** [comment]
Hello @frostblogNet,

I apologize for the delay in getting back to you on this issue.  We've done some digging, and have both identified a potential area for working around this, as well as an opportunity for us to improve this in the future.

This issue is caused primarily by the fact that as of Magento 2.3.1 (specific commit [here](https://github.com/magento/magento2/commit/76d478fa2c70c5c6ac07c7ef781ce46aeadffd06)), logged in users are not allowed to initiate password reset requests.

Our recommended fix, and what we will be working to include in a future release is simply removing the link on the customer information page, and replacing it with some text about needing to reset passwords on the sign-in page instead.

I'm going to go ahead and keep this issue open for now, but will make sure we track it as waiting for release.  Please feel free to update or respond to this issue if you have any more questions or concerns.

**Comment by christianzichichi** [comment]
Hi @frostblogNet,
We apologize for the delay here. A fix for this issue has been included in version 3.4.1 of our plugin. this plugin version will be included in one of the future Magento 2 releases. patch instructions to upgrade and try it out in the current Magento release (2.3.4) and in earlier ones will be released very soon here: https://github.com/amzn/amazon-payments-magento-2-plugin/blob/master/PATCH_INSTRUCTIONS.MD

thanks
regards

Christian
